src/integrations/pathManager.ts

```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import paths from '@/utils/paths';
+import { manageLinesInFile } from '@/integrations/manageLinesInFile';
+
+let pathManager: ManualPathManager;
+
+// PathManager is the interface that anything that manages the
+// PATH variable must implement.
+interface PathManager {
+  // Makes real any changes to the system. Should be idempotent.
+  enforce(): Promise<void>
+  // Removes any changes that the PathManager may have made.
+  // Should be idempotent.
+  remove(): Promise<void>
+}
+
+// ManualPathManager is for when the user has chosen to manage
+// their PATH themselves. It does nothing.
+class ManualPathManager implements PathManager {
+  async enforce(): Promise<void> {}
+  async remove(): Promise<void> {}
+}
+
+// RcFilePathManager is for when the user wants Rancher Desktop to
+// make changes to their PATH by putting lines that change it in their
+// shell .rc files.
+export class RcFilePathManager implements PathManager {
+  constructor() {
+    const platform = os.platform();
+
+    if (platform !== 'linux' && platform !== 'darwin') {
+      throw new Error(`Platform "${ platform }" is not supported by RcFilePathManager`);
+    }
+  }
+
+  async enforce(): Promise<void> {
+    await this.managePosix(true);
+    await this.manageCsh(true);
+    await this.manageFish(true);
+  }
+
+  async remove(): Promise<void> {
+    await this.managePosix(false);
+    await this.manageCsh(false);
+    await this.manageFish(false);
+  }
+
+  protected async managePosix(desiredPresent: boolean): Promise<void> {
+    const pathLine = `PATH=\${PATH}:${ paths.integration }`;
+
+    await Promise.all(['.bashrc', '.zshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
```

Comment:
Wouldn't this create a `~/.zshrc` even if I don't use zsh?

I'm not sure if that's good or not… Probably fine.

Reply:
Yes it does. As far as I can tell this is the best way to do it. A couple reasons:

- We could introduce some mechanism to detect whether the user has a shell installed, and only write the relevant rc file if it is there. However there are edge cases here, like what happens if a user installs a shell after RD runs, or what happens if they uninstall the shell. It's feasible to do, but I don't think it's worth it given that...

- `RcFilePathManager` is intended for the "beginner" use case. It needs to just work and there shouldn't be any hiccups. The fact that it is aimed at beginners also means that we aren't likely to annoy anybody by cluttering up their system with rc files for shells they don't use.

That's the way I was thinking about it. We could introduce more logic here, but this way of managing integrations was supposed to be simpler than our current way.